### PR TITLE
194-Send-Information-To-PostUrlComponent

### LINKED ISSUE #194: Make possibility to send information (title, description, url of uploaded file in firebase) from CV upload pop-up to PostUrlComponent
## User story

**Who:** **As a** developer

**What:** **I want to** get information from cv uploading pop-up including title, description and url of uploaded file into PostUrlComponent 
 
**Why:** **so that we can** So these information will be postet to the database

## Acceptance criteria

**Given** as an authenticated user

**When** Click on the Add Resume Button

## Implementation details

- [You define them here based on the above]

### PR DESCRIPTION
**Pull request by hamedveloper**
# Description

when the user clicls on the Add Resume buton, information about title, description and url have been delivered to the PostUrlComponent. 

Fixes #194 

# How to test?

npm run dev. and put Add Resume somewhere in API to test. 

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/components/AddResume/AddResume.js [hunk 1]
```diff
@@ -22,20 +23,24 @@ export const AddResume = () => {
...
       }
     }
   };
 
-  useStorage(uploadedFile);
+  const {url} = useStorage(uploadedMyFile);
+
+  const clickHandler = () => {    
+        PostUrlComponent({uploadedFile:url, title, description})
+      };
```
Comment (arya56) [inline src/client/components/AddResume/AddResume.js:43]: @pdsorensen @hamedveloper both component work properly. but when data send by the clickHandler we are getting this error: 
image

**Review by arya56 — COMMENTED** [review]
Here we getting this error.

File: src/client/components/AddResume/AddResume.js [hunk 2]
```diff
@@ -22,20 +23,24 @@ export const AddResume = () => {
...
+        setUploadedMyFile(null);
         setError('Please select a pdf file');
       }
     }
   };
 
-  useStorage(uploadedFile);
+  const {url} = useStorage(uploadedMyFile);
+
+  const clickHandler = () => {
```
Comment (pdsorensen) [inline src/client/components/AddResume/AddResume.js:41]: Close, but not quite. I think your reasoning is to think in props - that is one way of doing things, but there are better ways, and how you are thinking in hooks. 

That means we avoid passing props, and instead have shared "states". Here is an illustration, where you need to swap "store" with "my-hook". 

image

in other words, use the useStorage in posturlcomponent.

File: src/client/components/AddResume/AddResume.js [hunk 3]
```diff
@@ -4,12 +4,13 @@ import Button from '../Button/Button';
...
 import close from '../../assets/images/closeIcon.svg';
 import { useStorage } from '../../hooks/fileUploader';
+import PostUrlComponent from '../PostUrl/PostUrlComponent';
 
 export const AddResume = () => {
   const [title, setTitle] = useState('');
   const [description, setDescription] = useState('');
   const [error, setError] = useState(null);
-  const [uploadedFile, setUploadedFile] = useState('');
+  const [uploadedMyFile, setUploadedMyFile] = useState('');
```
Comment (pdsorensen) [inline src/client/components/AddResume/AddResume.js:13]: I think the naming before was better.

File: src/client/components/AddResume/AddResume.js [hunk 4]
```diff
@@ -22,20 +23,24 @@ export const AddResume = () => {
   const handleChange = (e) => {
     const selectedFile = e.target.files[0];
 
-    setUploadedFile(selectedFile);
+    setUploadedMyFile(selectedFile);
```
Comment (pdsorensen) [inline src/client/components/AddResume/AddResume.js:26]: Try to add this function to the hook instead.

**Review by pdsorensen — CHANGES_REQUESTED** [review]

File: src/client/components/AddResume/AddResume.js [hunk 5]
```diff
@@ -1,47 +1,60 @@
...
 export const AddResume = () => {
   const [title, setTitle] = useState('');
   const [description, setDescription] = useState('');
   const [error, setError] = useState(null);
-  const [uploadedFile, setUploadedFile] = useState('');
+  const { uploadedFile, setUploadedFile, uploadFile, uploadToStorage, url } = useStorage();
+  const { userData } = useAuthentication();
+  const history = useHistory();
+
+  // console.log(url);
```
Comment (pdsorensen) [inline src/client/components/AddResume/AddResume.js:19]: Remove these please.

File: src/client/hooks/fileUploader.js [hunk 6]
```diff
@@ -1,34 +1,61 @@
 import { useState, useEffect } from 'react';
```
Comment (pdsorensen) [inline src/client/hooks/fileUploader.js:1]: Remove useEffect

File: src/client/components/AddResume/AddResume.js [hunk 7]
```diff
@@ -1,47 +1,60 @@
...
 
   const handleChange = (e) => {
     const selectedFile = e.target.files[0];
-
     setUploadedFile(selectedFile);
 
     if (selectedFile) {
       if (types.includes(selectedFile.type)) {
         setError(null);
         setUploadedFile(selectedFile);
```
Comment (pdsorensen) [inline src/client/components/AddResume/AddResume.js:38]: I think we can remove setUploadedFile and uploaded file?

**Review by pdsorensen — COMMENTED** [review]
Fix static code build step and optionally try to implement the changes i asked for. 

Cheers,

File: src/client/components/AddResume/AddResume.js (hunk 7 above)
Comment (KavyaReddy732) [reply src/client/components/AddResume/AddResume.js:38]: we only can remove setUploadedFile inside the condition. if we remove them altogetherthen we are missing the hook which upload file into firebase.

**Review by pdsorensen — APPROVED** [review]
This isn't pretty, but it is a start. 

Cheers, kudos for efforts.

**Pull request merged**